index.rst

```diff
@@ -2,28 +2,137 @@
+
+.. _ToO_survey: https://rubin-sim.lsst.io/api/rubin_sim.scheduler.surveys.ToO_survey.html#rubin_sim.scheduler.surveys.ToO_survey
+
+With that in place, the only additional element required by the Scheduler is a source of ToOs events.
+
+ToO Events as Telemetry
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Currently the Scheduler CSC colects telemetry from the EFD and hands them over to the ``Driver``, which is in charge of formatting it in a way the scheduling algorithm understands.
+By using a general purpose interface for colecting and passing telemetry from the EFD to the scheduling algorithm, it allows us to easily add new data sources as long as the data is in the EFD.
+
+It is natural then to expect that, any new information needed by the Scheduler, should also be in the EFD.
+
+By writting ToO events to the EFD the Scheduler CSC will naturally be able to access them.
+Next, all that is required is for the ``Driver`` be able to parse the information in a way the scheduling algorithm understands.
+
+This design also has the advantage that it allows us to control which ToO events the Scheduler should consider by filtering which events are written to the EFD, or adding metadata to indicate whether an event was certified or not.
+
+The diagram below summarizes the design of the system.
+
+.. figure:: /_static/ToO_diagram.png
+   :name: fig-too-diagram
+   :target: ../_images/ToO_diagram.png
+   :alt: ToO automatic response system design.
+
+   Design of the system to autonomously handle ToO with the Scheduler CSC.
+
+The service in charge of reading/receiving ToO alerts from external sources is the ``ToO Alert Producer``, shown in the :ref:`diagram above <fig-too-diagram>`.
+This service will also be in charge of certifying/validating which ToO events the Scheduler will consider.
+
+Furthermore, the ``ToO Alert Producer`` can be designed in such a way to support multiple ``ToO Alert Stream`` sources.
+
+Manual Response With the ScriptQueue
+------------------------------------
+
+As mentioned above, in order for the Scheduler CSC to be able to respond to a certain type of ToO, it needs to be pre-configured with a setup capable of processing the incoming data stream and acting accordingly.
+This process should work on those cases where we know what to expect and are capable of planning ahead of time (e.g. Gravitational-wave events that got pre-approved by the project).
+
+Nevertheless, it might be that an important unplanned event occurs and requires immediate response, for which the Scheduler CSC won't be capable of handling.
+
+For those situations, a list of observations can be generated and added to the ScriptQueue.
+Since the Scheduler and the ScriptQueue are designed to seamlessly interoperate, it is possible the manually add a series of observations to the queue without worrying about the Scheduler.
+
+In these cases, the ScriptQueue will execute the observations, the Scheduler will wait for the execution to complete and then resume operations.
```

Comment:
(after moving line 111 upwards, then combine 112/114 into a new paragraph) -- 
"In these cases, the ScriptQueue will execute the urgent ToO observations. Since the Scheduler and ScriptQueue CSCs are designed to operate together, this manual addition of a series of observations to the queen simply causes the Scheduler CSC to wait for the ToO execution to complete. Then the Scheduler CSC will resume operations.